daemon/list.go

```diff
@@ -580,18 +581,45 @@ func includeContainerInList(container *container.Snapshot, filter *listContext)
 // refreshImage checks if the Image ref still points to the correct ID, and updates the ref to the actual ID when it doesn't
 func (daemon *Daemon) refreshImage(ctx context.Context, s *container.Snapshot, filter *listContext) (*types.Container, error) {
```

Comment:
Interesting; it looks like `filter` is unused. The argument is needed to match the `containerReducer` type/signature, but we should check if there's actually any implementations left that use the argument.

Might be good to change `filter` to `_` to make it clear it's not used here.

Reply:
Removed the `filter` arg completely, as `refreshImage` is the only `containerReducer`.

Reply:
ah, nice!

I see some cleaning opportunities up in the future 🪄 . Looks like `daemon.reduceContainers` is only called in one place. And that always calls it with `daemon.refreshImage` as argument, so we can probably get rid of all that indirection.

Definitely not needed for this PR, just an observation 👍

Reply:
- created https://github.com/moby/moby/issues/46082